docs/dev/adding_instruments.rst

```diff
@@ -196,15 +196,17 @@ If, however, you are unable to use the :py:class:`~pymeasure.adapters.VISAAdapte
 
 .. code-block:: python
 
-    def __init__(self, resourceName, baud_rate=2400, **kwargs):
+    def __init__(self, adapter, baud_rate=2400, **kwargs):
         kwargs.setdefault('timeout', 0.5)
         kwargs.setdefault('xonxoff', True)
-        adapter = SerialAdapter(resourceName, 
+        name = kwargs.pop('name', "Extreme 5000")
+        adapter = SerialAdapter(adapter, 
```

Comment:
@bmoneke, I am sorry for late comment, especially because it is not related to your changes.
I think this section is not so relevant and I have difficulty to find a matching use case.
In my understanding, the controller (PC) uses the `Adapter` to talk to instrument and there are two cases:
- User passes an address (string) and this is translated into a `VISAAdapter` instance 
- User pass it's own `Adapter` subclass instance that identifies the adapter being used.

An instrument driver developer should not make any assumption regarding the interface being used, since this is a user choice.

I would propose to remove this section if there is an agreement on this

Reply:
Some instruments have their own adapter, which is specific to the Instrument (in most cases it should be possible to move these implementations to the instrument, that is what I try in #660 ).

Removing this section prevents people from writing their own adapters, which is good, in my opinion, so I'm with you removing this section.

Reply:
@bmoneke, I have to say that I am not against people using their own adapters. For example, if you have a serial instrument, you can use a `VISAAdapter` or `SerialAdapter`.
The first one is selected by user specifying a string address, but I can also do:
```
my_adapter = SerialAdapter (...)
inst = InstClass(adapter=my_adapter)
```
I just don't see the use case where the `Adapter` is imposed by the instrument driver writer.
I also agree 100% with your comment that instrument behaviour should not be encapsulated in a custom adapter

Reply:
> Removing this section prevents people from writing their own adapters, which is good, in my opinion, so I'm with you removing this section.

I meant preventing the instrument authors to write specific adapters for their instruments and probably encapsulating these adapters in the instrument init. The reason, why I don't like this kind of adapter use (imposed by the instrument author), is exactly what you say, that you cannot use another adapter (e.g. our ProtocolTest) or Serial instead of pyvisa etc. in this case.

In #660 I try to remove all those special adapters enforced in `instrument.__init__`s in order to allow different (generally defined or personal) adapters to be used.

Reply:
> > Removing this section prevents people from writing their own adapters, which is good, in my opinion, so I'm with you removing this section.
> 
> I meant preventing the instrument authors to write specific adapters for their instruments and probably encapsulating these adapters in the instrument init. The reason, why I don't like this kind of adapter use (imposed by the instrument author), is exactly what you say, that you cannot use another adapter (e.g. our ProtocolTest) or Serial instead of pyvisa etc. in this case.
> 
> In #660 I try to remove all those special adapters enforced in `instrument.__init__`s in order to allow different (generally defined or personal) adapters to be used.

:+1: on this. This is relevant when you need to force use of specific/special adapters in an instrument. Dealing with this (as Benedikt says, ideally removing this) should be dealt with in #660, where this is better in scope.